Formula/rdm.rb

```diff
@@ -0,0 +1,97 @@
+    sha256 "35a4ad968ca1f30db62fc98dbb8434e86ddad04fb4f6d238abde04d6a92335c5"
+  end
+
+  resource "gbreakpad" do
+    url "https://github.com/google/breakpad/archive/9fecc95c72549452959431ddc0e4ec4e0cda8689.tar.gz"
+    sha256 "380137834d4c1d0e5f8ece4f1a7a27bfe250a123c607e2a8572be41bbb839335"
+    patch do
+      url "https://chromium-review.googlesource.com/changes/breakpad%2Fbreakpad~1338279/revisions/2/patch?zip"
+      sha256 "21b71807c4093b921744340e4b79d8bbb46504cd6ead416dc5a242c4191b71c1"
+    end
+  end
+
+  resource "hiredis" do
+    url "https://github.com/redis/hiredis/archive/685030652cd98c5414ce554ff5b356dfe8437870.tar.gz"
+    sha256 "e3dd3d13ae5156d1ab3004b082658d3ad8306881b7d940801bc3102b28d4915d"
+  end
+
+  resource "qredisclient" do
+    url "https://github.com/uglide/qredisclient/archive/42904d8fe83805c7c850ffaad25939106592efbd.tar.gz"
+    sha256 "3e1a9a1235141d911e8e02ed1e0be177c600ed7be230ecac3a57e264a2af64e4"
+  end
+
+  def install
+    nested_resources = {}
+    resources.each do |resource|
+      if resource.name == "hiredis"
+        nested_resources["qredisclient"] = resource
+      else
+        resource.stage buildpath/"3rdparty"/resource.name
+      end
+    end
+    nested_resources.each do |parent, resource|
+      resource.stage buildpath/"3rdparty"/parent/"3rdparty"/resource.name
+    end
+    ENV.prepend_path "PATH", Formula["make"].opt_libexec/"gnubin"
+    cd "3rdparty" do
+      chmod 0755, "gbreakpad"
+      cd "gbreakpad" do
+        args = %W[
+          -project src/client/mac/Breakpad.xcodeproj
+          -configuration Release
+          -target Breakpad
+          -xcconfig ../../src/gbreakpad.xcconfig
+          ARCHS=x86_64
+          MACOSX_DEPLOYMENT_TARGET=#{MacOS.version}
+        ]
+        xcodebuild "-sdk", MacOS.sdk_path, "SYMROOT=build", "DSTROOT=#{prefix}", "install", *args
+      end
+    end
+    cd "3rdparty/crashreporter" do
+      app_name = "RedisDesktopManager"
+      crash_server_url = "http://oops.redisdesktop.com/crash-report"
+      app_version = "0.9.9"
+      system "#{Formula["qt"].opt_bin}/qmake",
+        "CONFIG+=release",
+        "DESTDIR=#{buildpath}/bin/osx/release",
+        "DEFINES+=APP_NAME=\\\\\\\"" + app_name + "\\\\\\\"",
+        "DEFINES+=CRASH_SERVER_URL=\\\\\\\"" + crash_server_url + "\\\\\\\"",
+        "DEFINES+=APP_VERSION=\\\\\\\"" + app_version + "\\\\\\\"",
+        "crashreporter.pro"
+      system "make"
+      system "make", "install"
+    end
+    cd "src" do
+      cp "resources/Info.plist.sample", "resources/Info.plist"
+      system "#{Formula["qt"].opt_bin}/qmake", "CONFIG-=debug", "CONFIG+=release", "rdm.pro"
+      system "make"
+      system "make", "install"
+    end
+    prefix.install "bin/osx/release/rdm.app"
+    bin.install_symlink prefix/"rdm.app/Contents/MacOS/rdm"
+  end
+
+  test do
+    system "#{bin}/rdm", "--version"
```

Comment:
This test is not acceptable. A test is required that actually exercises some functionality.

Reply:
I see a bunch of other formulas doing this. For example [here](https://github.com/Homebrew/homebrew-core/blob/f5b2a69e7659bfcbdeb3fb0bc5a7bd526b143158/Formula/tarsnap-gui.rb#L26). Isn't this enough to verify that there is an executable binary?

Reply:
Just because other formulas are doing it doesn't mean we want to accept new formulas using bad older conventions. If there is a *very* compelling case to add a new formula and a *very* compelling case as to why there is no possibility of more thorough testing, then we will *consider* adding a new formula without more advanced testing. However, unless you can demonstrate that there is no other feasible way to test for correct functionality, this test is a non-starter for us, and the formula won't be accepted unless the test is improved.

Reply:
These are the command line options:
```
$ rdm --help
Usage: rdm [options]

Options:
  -h, --help                              Displays this help.
  -v, --version                           Displays version information.
  --settings-dir <settingsDir>            (Optional) Directory where RDM
                                          looks/saves .rdm directory with
                                          connections.json file
  --formatters-dir <formattersDir>        (Optional) Directory where RDM looks
                                          for native value formatters
  --rendering-backend <renderingBackend>  (Optional) QML rendering backend
                                          [software|opengl|d3d12|'']
```

Reply:
You may be able to spin up an instance of the app with the `--settings-dir`, option set, then send a signal to shutdown and then check for the existence, and, possibly the contents, of the `connections.json` file. You can also look for some correct logging to stdout/stderr, if it's suitably general and not likely to change between versions, or at any log files written elsewhere by the app bundle.